lib/ipmi_user.c

```diff
@@ -626,42 +668,43 @@ ipmi_user_mod(struct ipmi_intf *intf, int argc, char **argv)
 	return 0;
 }
 
-#define USER_PW_IPMI15_LEN 16 /* IPMI 1.5 only allowed for 16 bytes */
-#define USER_PW_IPMI20_LEN 20 /* IPMI 2.0 allows for 20 bytes */
-#define USER_PW_MAX_LEN USER_PW_IPMI20_LEN
-
 int
 ipmi_user_password(struct ipmi_intf *intf, int argc, char **argv)
 {
-	char *password = NULL;
+	char *password;
+	char passbuf[USER_PW_MAX_LEN + 1];
+	char tmp[USER_PW_MAX_LEN + 1];
 	int ccode = 0;
 	uint8_t password_type = USER_PW_IPMI15_LEN;
 	size_t password_len;
 	uint8_t user_id = 0;
+	size_t tmplen;
+
 	if (is_ipmi_user_id(argv[2], &user_id)) {
 		return (-1);
 	}
 
 	if (argc == 3) {
 		/* We need to prompt for a password */
-		char *tmp;
-		size_t tmplen;
-		password = ask_password(user_id);
-		if (!password) {
-			lprintf(LOG_ERR, "ipmitool: malloc failure");
+		memset(passbuf, 0, sizeof(passbuf));
+		memset(tmp, 0, sizeof(tmp));
+
+		if (ask_password(user_id, passbuf, sizeof(passbuf))) {
```

Comment:
Above you're memsetting the password buffer in all cases, but not here.

Reply:
input buffer isn't touched in the failure case.

Reply:
Marking resolved since no feedback was returned.